### PR TITLE
Ruby 3.0 upgrade

### PR DESCRIPTION
**Pull request by lazydevorg**
### Context

- Ticket: CST-153

### Changes proposed in this pull request

- Ruby upgrade from 2.7 to 3.0
- Rubocop fixes

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -28,31 +28,31 @@ WORKDIR /app
...
+RUN bundler -v
+RUN bundle config set no-cache 'true'
+RUN bundle config set no-binstubs 'true'
+RUN bundle --retry=5 --jobs=4 --without=development test
+RUN yarn install --check-files --production
+RUN apk del .gem-installdeps
+RUN rm -rf /usr/local/bundle/cache
+RUN find /usr/local/bundle/gems -name "*.c" -delete
+RUN find /usr/local/bundle/gems -name "*.h" -delete
+RUN find /usr/local/bundle/gems -name "*.o" -delete
```
Comment (StupidCodeFactory) [inline Dockerfile:46]: you'd want to run all of those in one in single comment, hence the previous `&& \` in order for docker to cache the layer adequately.

File: app/forms/participants/participant_validation_form.rb [hunk 2]
```diff
@@ -48,7 +48,7 @@ def self.call(participant_profile, save_validation_data_without_match: true, dat
 
     step :trn, update: true do
       attribute :trn, :string
-      attribute :no_trn, :boolean, default: false
+      attribute :no_trn, :boolean
```
Comment (StupidCodeFactory) [inline app/forms/participants/participant_validation_form.rb:51]: what's the rational behind removing the default?

File: app/lib/multistep/form.rb [hunk 3]
```diff
@@ -66,13 +66,13 @@ def completed_steps=(value)
     end
 
     def step_completed?(step)
-      completed_steps.include?(step)
+      completed_steps&.include?(step)
```
Comment (StupidCodeFactory) [inline app/lib/multistep/form.rb:69]: Would you be able to provide some context as to why `completed` can now be `nil`?

**Comment by lazydevorg** [comment]
a new PR has been opened to upgrade to Ruby 3.1 (https://github.com/DFE-Digital/early-careers-framework/pull/2188)

**Pull request closed without merging**